c/meterpreter/source/extensions/powershell/powershell_bridge.h

```diff
@@ -5,12 +5,12 @@
 #ifndef _METERPRETER_SOURCE_EXTENSION_POWERSHELL_BRIDGE_H
 #define _METERPRETER_SOURCE_EXTENSION_POWERSHELL_BRIDGE_H
 
-DWORD initialize_dotnet_host();
+HRESULT initialize_dotnet_host();
 VOID deinitialize_dotnet_host();
-DWORD request_powershell_execute(Remote *remote, Packet *packet);
-DWORD request_powershell_shell(Remote *remote, Packet *packet);
-DWORD request_powershell_session_remove(Remote *remote, Packet *packet);
-DWORD request_powershell_assembly_load(Remote *remote, Packet *packet);
-DWORD invoke_startup_script(LPCSTR script);
+HRESULT request_powershell_execute(Remote *remote, Packet *packet);
+HRESULT request_powershell_shell(Remote *remote, Packet *packet);
+HRESULT request_powershell_session_remove(Remote *remote, Packet *packet);
+HRESULT request_powershell_assembly_load(Remote *remote, Packet *packet);

```

Comment:
So this technically seems okay right now because the [dispatch loop](https://github.com/rapid7/metasploit-payloads/blob/master/c/meterpreter/source/metsrv/base.c#L327) doesn't do anything with the result however I'm pretty sure that all the other request handlers return a DWORD that's a Win32 error. Returning an HRESULT instead seems like the type of thing that could be problematic in the future. I guess it's not documented though in the [DISPATCH_ROUTINE](https://github.com/rapid7/metasploit-payloads/blob/c3b9bbc18889f952b65d1905cc07424141dc86f3/c/meterpreter/source/common/common_base.h#L11) definition though.

Reply:
yea I agree it's not ideal but I wasn't sure what else I could reasonably do, anyone has any ideas/thoughts though I'd be happy to implement them